src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/UriSearchValue.cs

```diff
@@ -38,43 +88,58 @@ public UriSearchValue(string uri)
         /// Parses the string value to an instance of <see cref="UriSearchValue"/>.
         /// </summary>
         /// <param name="s">The string to be parsed.</param>
+        /// <param name="separateCanonicalComponents">When true, the Version and Fragment will be separated into Canonical components</param>
+        /// <param name="modelInfoProvider">FHIR Model Info provider</param>
         /// <returns>An instance of <see cref="UriSearchValue"/>.</returns>
-        public static UriSearchValue Parse(string s)
+        public static UriSearchValue Parse(string s, bool separateCanonicalComponents, IModelInfoProvider modelInfoProvider)
         {
             EnsureArg.IsNotNullOrWhiteSpace(s, nameof(s));
+            EnsureArg.IsNotNull(modelInfoProvider, nameof(modelInfoProvider));
+
+            if (modelInfoProvider.Version == FhirSpecification.Stu3 || !_canonicalFormat.IsMatch(s))
+            {
+                return new UriSearchValue(s, false);
+            }
 
-            return new UriSearchValue(s);
+            return new UriSearchValue(s, separateCanonicalComponents);
         }
 
         /// <inheritdoc />
-        public void AcceptVisitor(ISearchValueVisitor visitor)
+        public virtual void AcceptVisitor(ISearchValueVisitor visitor)
         {
             EnsureArg.IsNotNull(visitor, nameof(visitor));
 
             visitor.Visit(this);
         }
 
-        public bool Equals([AllowNull] ISearchValue other)
+        public virtual bool Equals([AllowNull] ISearchValue other)
         {
-            if (other == null)
-            {
-                return false;
-            }
-
             var uriSearchValueOther = other as UriSearchValue;
 
             if (uriSearchValueOther == null)
             {
                 return false;
             }
 
-            return Uri == uriSearchValueOther.Uri;
+            return ToString() == uriSearchValueOther.ToString();
```

Comment:
Are the canonical components case sensitive for the comparison?  Should this be a case insensitive comparison?

Reply:
It does say: `URLs are always considered to be case-sensitive` (https://www.hl7.org/fhir/references.html#literal)
However, it probably helps to be explicit about this.
```suggestion
            return string.Equals(ToString(), uriSearchValueOther.ToString(), StringComparison.Ordinal);
```

Reply:
Updated